### DISCUSSION TITLE
Add a community extra repo like astrocommunity

### DISCUSSION THREAD

**Discussion by dragove** [LazyVim/LazyVim#686] [Ideas]
There are some plugins in lazyvim/plugins/extras, maybe it is better to create a new repo just like https://github.com/AstroNvim/astrocommunity and then everyone can simply use common configs just like vscodes's plugin market place.

**Comment by albohlabs** [comment]
I like the idea a lot. Maybe this is something the LazyVim community can do too. I could imagine something like `LazyVim-extras`, where LazyVim plugins/extras can find a central place to be managed and discussed.

**Comment by nipuna-perera** [comment]
Love this idea! I can see it being very helpful to pull in language based
setups. All alterations can be documented in the repo too.

**Comment by rmunn** [comment]
It might also be possible to just use the https://github.com/AstroNvim/astrocommunity repo directly in one's own LazyVim config. Here's how.

First, the way AstroNvim works is that its setup imports a `community.lua` file from the user's config. Then the next thing it does is import the `plugins` folder from the user's config. And their documentation suggests importing plugins in `community.lua` and then overriding their config in the `plugins` folder, as follows:

```lua
-- This would be in the user's `community.lua` file:
return {
  "AstroNvim/astrocommunity",
  { import = "astrocommunity.colorscheme.catppuccin" },
  -- ... other plugins ...
}
```

Then they suggest overriding the community's default settings via something like this:

```lua
-- This would be in the user's `plugins/catppuccin.lua` file:
return {
  { -- further customize the options set by the community
    "catppuccin",
    opts = {
      integrations = {
        sandwich = false,
        noice = true,
        mini = true,
        leap = true,
        markdown = true,
        neotest = true,
        cmp = true,
        overseer = true,
        lsp_trouble = true,
        rainbow_delimiters = true,
      },
    },
  },
}
```

Translated into LazyVim terms, you would do the following. First, create a file like `plugins/astrocommunity-init.lua`:

```lua
return {
  "AstroNvim/astrocommunity",
  { import = "astrocommunity.colorscheme.catppuccin" },
  -- ... other plugins ...
}
```

Then in `plugins/astrocommunity-overrides.lua` (or any other filename that sorts alphabetically **later** than the first one):

```lua
return {
  {
    "catppuccin",
    opts = {
      -- override catppuccin options here
    }
  }
}
```

I have not tested this, so there might be issues with some AstroNvim community configs, e.g. if they assume the existence of some plugins that are in AstroNvim by default but not in LazyVim. However, there's no reason why LazyVim users shouldn't be able to make use of AstroNvim community plugins; it appears to be rather simple overall. (If I try it and find that it's not so simple, I'll come back and edit this comment).
